contracts/serverless-v2/Jobs.sol

```diff
@@ -411,29 +482,52 @@ contract Jobs is
 
         address gateway = jobs[_jobId].gateway;
         address jobOwner = jobs[_jobId].jobOwner;
-        bool isNoOutputSubmitted = (jobs[_jobId].outputCount == 0);
+        uint8 outputCount = jobs[_jobId].outputCount;
+        bool isNoOutputSubmitted = (outputCount == 0);
+        uint256 deadline = jobs[_jobId].deadline;
+        uint256 firstOutputExecutionTime = jobs[_jobId].firstOutputExecutionTime;
         delete jobs[_jobId];
 
+        _releaseEscrowAmount(gateway, outputCount, isNoOutputSubmitted, deadline, firstOutputExecutionTime);
+
         // slash Execution node
         uint256 len = selectedExecutors[_jobId].length;
         for (uint256 index = 0; index < len; index++) {
             address executor = selectedExecutors[_jobId][index];
-            executors.slashExecutor(
-                executor, 
-                hasExecutedJob[_jobId][executor], 
-                isNoOutputSubmitted,
-                gateway,
-                jobOwner
-            );
-
-            if(!hasExecutedJob[_jobId][executor])
+
+            if(!hasExecutedJob[_jobId][executor]) {
+                executors.slashExecutor(
+                    executor,
+                    isNoOutputSubmitted,
+                    gateway,
+                    jobOwner
+                );
                 emit SlashedOnExecutionTimeout(_jobId, executor);
+            }
             delete hasExecutedJob[_jobId][executor];
         }
 
         delete selectedExecutors[_jobId];
     }
 
+    function _releaseEscrowAmount(
+        address _gateway,
+        uint8 _outputCount,
+        bool _isNoOutputSubmitted,
+        uint256 _deadline,
+        uint256 _firstOutputExecutionTime
+    ) internal {
+        // transfer back the whole escrow amount to gateway if no output submitted
+        if(_isNoOutputSubmitted)
+            TOKEN.safeTransfer(_gateway, _deadline * (EXECUTOR_FEE_PER_MS + STAKING_REWARD_PER_MS));
+        else if(_outputCount == 1) {
+            uint256 executor1Payout = (_firstOutputExecutionTime * EXECUTOR_FEE_PER_MS * 2) / 3;
+            uint256 gatewayDeposit = _deadline * (EXECUTOR_FEE_PER_MS + STAKING_REWARD_PER_MS);
+            uint256 gatewayPayout = gatewayDeposit - STAKING_REWARD_PER_MS - executor1Payout;
+            TOKEN.safeTransfer(_gateway, gatewayPayout);
+        }
```

Comment:
realized, I missed the third case, where 3rd output hasn't submitted, need to return remaining deposit in this case too. Added it to the doc too. Please add that handling.

Reply:
done